### PR TITLE
[hotfix]Harden TextInputFormatTest to avoid failure by uncleaned files

### PR DESCRIPTION
**Pull request by Aitozi**
## What is the purpose of the change

When I run test locally, it happened to failed and with the uncleaned temp file leaving in the tmp dir. So the test case can not pass ever until i delete the correspond files.

It use `deleteOnExit` to clean up the temp file created by test case, But the method may not work when JVM crash or killed. I think it may make the test case fragile.   

## Brief change log

Add the @Before and @After prepare / cleanup method

### CODE REVIEW

**Comment by Aitozi** [comment]
And I see there are quite a lot other tests rely on `deleteOnExit` to do clean up. Should I do the fix for them, too?  Please take a look when you free , Thanks @zentol .

File: flink-java/src/test/java/org/apache/flink/api/java/io/TextInputFormatTest.java [hunk 1]
```diff
@@ -44,14 +48,34 @@
  * Tests for {@link TextInputFormat}.
  */
 public class TextInputFormatTest {
+
+	private File baseDir;
+	private final String tempDirName = "tempDir";
+
+	@Before
+	public void setUp() throws IOException {
```
Comment (zentol) [inline flink-java/src/test/java/org/apache/flink/api/java/io/TextInputFormatTest.java]: use a `TemporaryFolder` instead

Comment (Aitozi) [reply flink-java/src/test/java/org/apache/flink/api/java/io/TextInputFormatTest.java]: Seems better, I will change it then. But a little question, why my approach can not ensure the file cleaned (I check and clean the related file before execute and after test method), Thanks.

Comment (zentol) [reply flink-java/src/test/java/org/apache/flink/api/java/io/TextInputFormatTest.java]: if the JVM crashes before the delete calls are executed the files will remain.

**Comment by Aitozi** [comment]
I update the PR, please take a look when you are free, thanks @zentol

**Pull request merged**